subprojects/core-api/src/main/java/org/gradle/StartParameter.java

```diff
@@ -176,7 +176,9 @@ public void setProjectCacheDir(@Nullable File projectCacheDir) {
     }
 
     /**
-     * Returns the project's cache dir.
+     * Returns the project's cache dir. Note that this directory is managed by Gradle, and plugins or build logic should not write directly to it.
```

Comment:
:x: I find the repetition between the two paragraphs to be a bit too much.

Let's keep the first paragraph simple and only add the ownership info in a second one.

```
/**
 * Returns the project's cache dir.
 * <p/>
 * Note that this directory is managed by Gradle, and it assumes full ownership of its contents.
 * Plugins and build logic should not store or modify any files or directories within this cache directory.
 *
```